app/back/src/Entity/UserAvatar.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace App\Entity;
+
+use ApiPlatform\Metadata\ApiProperty;
+use ApiPlatform\Metadata\ApiResource;
+use ApiPlatform\Metadata\Delete;
+use ApiPlatform\Metadata\Get;
+use ApiPlatform\Metadata\GetCollection;
+use ApiPlatform\Metadata\Post;
+use App\Controller\Api\UserAvatarController;
+use App\Entity\Traits\TimestampableTrait;
+use App\Repository\UserAvatarRepository;
+use DateTime;
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\HttpFoundation\File\File;
+use Symfony\Component\Serializer\Annotation\Groups;
+use Symfony\Component\Validator\Constraints as Assert;
+use Vich\UploaderBundle\Mapping\Annotation as Vich;
+
+#[ORM\Entity(repositoryClass: UserAvatarRepository::class)]
+#[ORM\HasLifecycleCallbacks]
+#[ApiResource(
+    operations: [
+        new GetCollection(),
+        new Get(
+            normalizationContext: ['groups' => ['avatar:read']],
+            security: 'is_granted("USER_PROFILE", object)',
+        ),
+        new Post(
+            inputFormats: ['multipart' => ['multipart/form-data']],
+            controller: UserAvatarController::class,
+            name: 'create_user_avatar',
+        ),
+        new Delete(
+            security: 'is_granted("USER_PROFILE", object)',
```

Comment:
```suggestion
            security: 'is_granted("USER_AVATAR_DELETE", object)',
```

Rajouter un VOTER `ùserAvatarVoter`

Reply:
creer  USER_AVATAR_DELETE et USER_AVATAR_CREATE